pkg/clusteragent/autoscaling/workload/controller_horizontal.go

```diff
@@ -106,22 +100,32 @@ func (hr *horizontalController) performScaling(ctx context.Context, podAutoscale
 		scaleDirection = scaleDown
```

Comment:
I think we should calculate the scaleDirection after applying `replicasWithRules`

Reply:
The scaleDirection cannot change from rules, we can only reduce the number of replicas.

Reply:
What if for example the targetDesiredReplicas is 3, currentReplica is 3 as well and the minReplica is 5 ?